### ISSUE TITLE
add option to keep either lat or lon dimension when sorting

### ISSUE THREAD

**Issue by MarvinLorenz** [florianboergel/pyTEF#41]
e.g. when providing not a transect, but a full 4D array, the option to keep either lon or lat should be possible, or automatically requested!
I think at the moment our code would simply sum over all spatial dimensions when not a transect is given.

**Comment by florianboergel** [comment]
OK. This is for the water mass transformation? Because otherwise it would be pretty straightforward, to explicitly mention that you have to preprocess your 4D data. Which means cut your transect using .sel before you intialize the tef object.

Our TEF object has the possibility to load an xarray.Dataset instead of load from a file. This would avoid 4D data.
`def __init__(self, filename = None, ds = None, **kwargs):`

**Comment by MarvinLorenz** [comment]
Yeah, it is useful if you want to see divergences in q with lon or lat for example. This then corresponds to the entrainment per salinity class, for example, along the estuary axis, see Wang et al. (2017), their eq. 2.6.

When I loaded 4D data I passed .isel(lon=[0]) to the sorting function which worked fine.

However, we should check in the sorting functions the dimensions if either lat or lon is of length 1. If none of the two is of length one, we should raise an error in the existing functions, since otherwise it we would just sum over which does not make any sense. 
If specified to keep either lon or lat, the dimension of the returned q and Q should be (time, salt, lon/lat). This would simply require an if statement in the creation of the output and another loop over lat/lon. Should be straightforward to code. It just has to be done. Could be maybe solved more elegantly somehow.
